internal/store/installation_dns.go

```diff
@@ -91,15 +91,42 @@ func (sqlStore *SQLStore) getDNSRecordsForInstallations(db queryer, installation
 
 // SwitchPrimaryInstallationDomain sets given domain as primary reducing all others to non-primary.
 func (sqlStore *SQLStore) SwitchPrimaryInstallationDomain(installationID string, installationDNSID string) error {
+	tx, err := sqlStore.beginTransaction(sqlStore.db)
+	if err != nil {
+		return errors.Wrap(err, "failed to start transaction")
+	}
+	defer tx.RollbackUnlessCommitted()
+
 	query := sq.Update(installationDNSTable).
-		Set("IsPrimary",
-			sq.Case().When(sq.Expr("ID = ?", installationDNSID), "true").Else("false"),
-		).Where("InstallationID = ?", installationID)
+		Set("IsPrimary", "false").
+		Where("installationID = ?", installationID)
 
-	_, err := sqlStore.execBuilder(sqlStore.db, query)
+	_, err = sqlStore.execBuilder(tx, query)
+	if err != nil {
+		return errors.Wrap(err, "failed to unset primary DNS for installation")
+	}
+
+	query = sq.Update(installationDNSTable).
+		Set("IsPrimary", "true").
+		Where("ID = ?", installationDNSID)
+
+	result, err := sqlStore.execBuilder(tx, query)
 	if err != nil {
 		return errors.Wrap(err, "failed to switch primary DNS")
 	}
+	// Perform validation that exactly one row was updated just in case.
+	count, err := result.RowsAffected()
+	if err != nil {
+		return errors.Wrap(err, "failed to count rows affected on IsPrimary=true update")
+	}
+	if count != 1 {
+		return errors.Errorf("expected exactly 1 row update, but got %d", count)
+	}
```

Comment:
The installationdns tables' ID column is a unique primary key so I think this check of one row edited will be redundant given you are using the ID column in the where clause

Reply:
I was mainly worried about 0 rows being edited. Starting to think this level of concern isn't necessary though.